compiler/rustc_hir/src/hir.rs

```diff
@@ -407,10 +407,8 @@ impl<'hir> PathSegment<'hir> {
 /// that are [just paths](ConstArgKind::Path) (currently just bare const params)
 /// versus const args that are literals or have arbitrary computations (e.g., `{ 1 + 3 }`).
 ///
-/// The `Unambig` generic parameter represents whether the position this const is from is
-/// unambiguously a const or ambiguous as to whether it is a type or a const. When in an
-/// ambiguous context the parameter is instantiated with an uninhabited type making the
-/// [`ConstArgKind::Infer`] variant unusable and [`GenericArg::Infer`] is used instead.
+/// For an explanation of the `Unambig` generic parameter see the dev-guide:
```

Comment:
That's a lot of links to the same guide section. Are they all necessary? Maybe just the ones on types, and cut the ones on methods? Feels like the balance between "thoroughness" and "don't repeat yourself" is off?

Reply:
Yeah, type-level linkage introducing the concepts the type's associated items reference makes more sense to me.